cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -34,20 +34,22 @@ type Reporter struct {
 	collector api_v2.CollectorServiceClient
 	logger    *zap.Logger
 	sanitizer zipkin2.Sanitizer
+	agentTags []model.KeyValue
 }
 
 // NewReporter creates gRPC reporter.
-func NewReporter(conn *grpc.ClientConn, logger *zap.Logger) *Reporter {
+func NewReporter(conn *grpc.ClientConn, agentTags string, logger *zap.Logger) *Reporter {
 	return &Reporter{
 		collector: api_v2.NewCollectorServiceClient(conn),
 		logger:    logger,
 		sanitizer: zipkin2.NewChainedSanitizer(zipkin2.StandardSanitizers...),
+		agentTags: parseAgentTags(agentTags),
 	}
 }
 
 // EmitBatch implements EmitBatch() of Reporter
 func (r *Reporter) EmitBatch(b *thrift.Batch) error {
-	return r.send(jConverter.ToDomain(b.Spans, nil), jConverter.ToDomainProcess(b.Process))
+	return r.send(addAgentTags(jConverter.ToDomain(b.Spans, nil), r.agentTags), jConverter.ToDomainProcess(b.Process))
```

Comment:
this statement is becoming hard to read, I recommend splitting into local vars

Reply:
Will do.